### PR TITLE
Add error trapping in Salvo test coverage checks

### PR DESCRIPTION
**Pull request by gyohuangxin**
The Salvo test and coverage CI workflows show green though there are errors: 
https://app.circleci.com/pipelines/github/gyohuangxin/envoy-perf/13/workflows/e1eb2506-0730-4012-9616-1fbbc9140927/jobs/47?invite=true#step-102-1050
https://app.circleci.com/pipelines/github/gyohuangxin/envoy-perf/13/workflows/e1eb2506-0730-4012-9616-1fbbc9140927/jobs/48?invite=true#step-102-1022

This PR aims to add trapping at above errors and fix them.

### CODE REVIEW

File: salvo/tools/coverage.sh [hunk 1]
```diff
@@ -35,6 +35,11 @@ CMD="${CMD} -o coverage/coverage.dat"
 
 # Extract all coverage data for salvo project files
 lcov -e coverage/coverage.dat -o coverage/salvo.dat '*/salvo/src/lib/*.py'
+if [ "$?" != "0" ]; then
```
Comment (mum4k) [inline salvo/tools/coverage.sh:38]: Checking the error after individual commands may still be error prone (we may forget to add checks when we add new commands).

Can we do it at the top of the file instead? This file could start as:

```
#!/bin/bash

# Exit immediately if a command exits with a non-zero status.
# pipefail indicates that the return value of a pipeline is the status
# of the last command to exit with a non-zero status.
set -eo pipefail
```

We may want to adopt this approach in all our CI scripts.

**Review by mum4k — CHANGES_REQUESTED** [review]
Thank you for spotting this problem.

File: salvo/tools/coverage.sh (hunk 1 above)
Comment (gyohuangxin) [reply salvo/tools/coverage.sh]: @mum4k Thanks for pointing out this better approach.

**Comment by gyohuangxin** [comment]
@mum4k It always fails with the error: https://app.circleci.com/pipelines/github/gyohuangxin/envoy-perf/18/workflows/1a177034-18ed-43f1-8094-ed51d7c45c28/jobs/65?invite=true#step-102-1006, and I think it related to OOM. Do you have any idea about how to resolve it so that I can verify code changes?

**Comment by mum4k** [comment]
@gyohuangxin running out of memory is the primary reason why we are switching away from CircleCI, we have more resources on AZP.

If memory is the bottleneck here, we may not be able to get this to work on CircleCI, or it might be throw-away work. Should we focus on the AZP migration first? We can disable portions of the CI script that cannot pass on CircleCI and then re-enable them once we are on AZP.

**Comment by gyohuangxin** [comment]
@mum4k Sorry for the late reply due to holiday. Sure, I will focus on the migration first.

**Comment by gyohuangxin** [comment]
@mum4k  Do you know how to invoke AZP retest without a new commit?

**Comment by mum4k** [comment]
@gyohuangxin repokkiteh is currently configured to [retest via CircleCI](https://github.com/envoyproxy/envoy-perf/blob/ad15bca77f7f450705981081bc194de175716dd6/repokitteh.star#L6). We will have to switch it to AZP, see this [example from Nighthawk](https://github.com/envoyproxy/nighthawk/blob/f911358d0e453180b3cc0cd8affe4b3319cfe6f9/repokitteh.star#L4-L6).

Once we do that, I will also need to define the secret token for repokkiteh to use when calling the AZP APIs. Personally I would keep this to the end and retry by adding commits for now.

**Comment by gyohuangxin** [comment]
@mum4k Thanks, understood.

**Comment by gyohuangxin** [comment]
@mum4k Ready for review.

**Review by mum4k — APPROVED** [review]

**Pull request merged**